site/content/integrate/apps/quick-start-serverless/_index.md

```diff
@@ -0,0 +1,443 @@
+				"name": "hello"
+			}
+		]
+	}
+}
+```
+
+### 2. Static (`./static`) 
+
+The example app will use an icon, you can download an example icon using:
+
+```bash
+mkdir ./static
+curl https://github.com/mattermost/mattermost-plugin-apps/raw/master/examples/go/hello-serverless/static/icon.png -o static/icon.png
+```
+
+### 3. `./function` package
+
+#### `bindings.json`
+```json
+{
+	"type": "ok",
+	"data": [
+		{
+			"location": "/command",
+			"bindings": [
+				{
+					"icon": "icon.png",
+					"label": "hello-serverless",
+					"description": "Hello Serverless app",
+					"hint": "[send]",
+					"bindings": [
+						{
+							"location": "send",
+							"label": "send",
+							"call": {
+								"path": "/send"
+							}
+						}
+					]
+				}
+			]
+		}
+	]
+}
+```
+
+#### `send_form.json`
+
+```json
+{
+	"type": "form",
+	"form": {
+		"title": "Hello, serverless!",
+		"icon": "/static/icon.png",
+		"fields": [
+			{
+				"type": "text",
+				"name": "message",
+				"label": "message"
+			}
+		],
+		"call": {
+			"path": "/send"
+		}
+	}
+}
+```
+
+### `function.go`
+
+```go
+package function
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-plugin-apps/apps/appclient"
+	"github.com/mattermost/mattermost-plugin-apps/utils/httputils"
+)
+
+//go:embed bindings.json
+var bindingsData []byte
+
+//go:embed send_form.json
+var formData []byte
+
+// Handler for OpenFaaS and faasd.
+func Handle(w http.ResponseWriter, r *http.Request) {
+	InitApp(apps.DeployOpenFAAS)
+	http.DefaultServeMux.ServeHTTP(w, r)
+}
+```
+
+Note: names `package function` and `func Handle` appear to be required
+(hardcoded) for OpenFaaS.
+
+`golang-middleware` template makes use of adding boilerplate handlers to
+`http.DefaultServeMux`, so we just need to add our handlers and serve.
+
+```go
+var deployType apps.DeployType
+
+func InitApp(dt apps.DeployType) {
+	// Returns the Channel Header and Command bindings for the App.
+	http.HandleFunc("/bindings", httputils.HandleJSONData(bindingsData))
+
+	// The form for sending a Hello message.
+	http.HandleFunc("/send/form", httputils.HandleJSONData(formData))
+
+	// The main handler for sending a Hello message.
+	http.HandleFunc("/send/submit", send)
+
+	deployType = dt
+}
+
+func send(w http.ResponseWriter, req *http.Request) {
+	creq := apps.CallRequest{}
+	json.NewDecoder(req.Body).Decode(&creq)
+
+	message := fmt.Sprintf("Hello from a serververless app running as %s!", deployType)
+	v, ok := creq.Values["message"]
+	if ok && v != nil {
+		message += fmt.Sprintf(" ...and %s!", v)
+	}
+
+	asBot := appclient.AsBot(creq.Context)
+	asBot.DM(creq.Context.ActingUserID, message)
+
+	httputils.WriteJSON(w,
+		apps.NewOKResponse(nil, "Created a post in your DM channel."))
+}
+```
+
+#### `go.mod` and `go.sum`
+
+Since package `function` is built independently by OpenFaaS, add its own `go.mod` for its ddependencies:
```

Comment:
Instead of using multiple `go.mod` files is it suitable to use a single one?

Reply:
It'd then be deploying the aws packages to openfaas build, etc, no? Also, one would need to copt it explicitly into `function` when building with OpenFaaS and Kubeless; what would the module name be?

Reply:
We can still have multiple `main`'s, if that is the requirement. The module name should just be `{{path-to-your-module}}`. `function` would just be a package that get's imported. Each deployment type get's there own `main`, but only the root directly contains a `go.mod`.

Reply:
Why does `function` need a seperate `go.mod`?